### DISCUSSION TITLE
[16.0.0-RC5] HA Energy dashboard Last reset missing warning (regression?)

### DISCUSSION THREAD

**Discussion by eugeniumegherea** [jomjol/AI-on-the-edge-device#3456] [General] [answered]
Hello, first of all thanks for the amazing project! Release v16 RC5 works brilliantly well for me so far, but I had slight issue with it.

Before opening an issue, would like to ask if this is maybe a regression with home assistant integration? Will keep digging and will update this post if I find anything relevant. It worked fine with v15.7.
![image](https://github.com/user-attachments/assets/846a648d-f27b-4692-9efc-a69efec053db)

This warning prevents energy consumption to plot it and to calculate totals.

I am inclined to think this is most likely weird behavior from HA but cannot confirm this yet. What I can tell is that the json interface  sent with MQTT didn't change between releases, see pictures, first picture is from 15.7 and last picture is from v16 rc5
![image](https://github.com/user-attachments/assets/ba319aed-6e9b-4e83-b395-bb679e1b960c)
![image](https://github.com/user-attachments/assets/b9e56e2d-8717-4ff1-9a35-36e66a83afe1)

Would appreciate any help with this issue. Thanks and happy holidays! 

Edit & Findings:
https://community.home-assistant.io/t/last-reset-is-missing-even-though-its-provided-in-config/435530/2
The unit I am using is of state_class measurement which makes sense having the response from link above in mind
![image](https://github.com/user-attachments/assets/28245f82-7a26-4e97-8c13-a501e3402eb2)

Was looking in diff between rc5 and 15.7 and saw those lines https://github.com/jomjol/AI-on-the-edge-device/compare/v15.7.0...v16.0.0-RC5?diff=split&w=#diff-11b6733951fb7b2cc28de71943598d8c180539e00880e3fa1e91fae0423d7438R195 and I think those are the reason why it changed. I am using allow negative rate because sometimes it confuses numbers and I would like it to correct itself and have negative numbers in the next measurements so that in statistic it ends up correct anyway (please correct me if I don't understand the use case for this feature)

Edit 2:
So I can confirm that disabling allow negative rate fixed this issue! I am not sure about the usefulness of this feature then. It has the potential to break other existing deployments of people in current state

**Comment by caco3** [comment, marked as answer]
Which version of Homeassistant are you using?
The change you are relating to is here: https://github.com/jomjol/AI-on-the-edge-device/pull/3332
It got required after Homeassistant `2024.10`, see https://github.com/jomjol/AI-on-the-edge-device/issues/3331

See also https://developers.home-assistant.io/blog/2021/08/16/state_class_total which clearly states that a value of a device which can decrease must not be `total_increasing`.

**Reply by eugeniumegherea** [reply]
Thanks for confirming my findings. I am on the latest (2024.12.5), running it inside a VM. I understand better now the required changes. It all makes total sense. My only remaining concern is that it might silently break energy for people who have allow negative enabled. But I think with a nice release note you could warn people

I will mark this discussion as answered, Thanks again for your hard work

**Comment by thomascandrian** [comment]
Hello,

I occasionally encounter faulty readings that push the actual value higher. These cannot be detected by the rate limit (in cases where some readings were not retrieved, while simultaneously much water was consumed, it will be a high rate that blocks readings forever, so I had to set it high enough, actually at 800L/min at 3 Minute per round. If I use 100L/min and 2 readings fail, the third will already reach that limit and prevent any subsequent higher reading).

I was hoping that the option for negative values would automatically correct this, as a negative value would lower the calculated rate in HA and balance out the error on the next run. Unfortunately, this does not seem to work as expected without the ability to set a reset date.

At the moment, I have disabled the negative value option, but I wanted to suggest that this feature could potentially be added in future versions as a configurable setting.

Thank you for your efforts!
